CMakeLists.txt

```diff
@@ -0,0 +1,123 @@
+endif ()
+
+# Download and extract Boost
+set(BOOST_VERSION "1.85.0" CACHE STRING "Version of Boost library to be used when compiling CLP sources.")
+string(REPLACE "." "_" BOOST_ARCHIVE_VERSION_PART ${BOOST_VERSION})
+
+set(BOOST_SRC_DIR "${CMAKE_CURRENT_BINARY_DIR}/boost" CACHE STRING
+    "Directory containing Boost sources, fetched from official release if non existent.")
+if (NOT EXISTS ${BOOST_SRC_DIR})
+    # Sets DOWNLOAD_EXTRACT_TIMESTAMP=ON in FetchContent_Declare() for CMake 3.24 and above
+    cmake_policy(SET CMP0135 NEW)
+
+    FetchContent_Declare(
+        Boost
+        URL "https://archives.boost.io/release/${BOOST_VERSION}/source/boost_${BOOST_ARCHIVE_VERSION_PART}.tar.gz"
+        SOURCE_DIR ${BOOST_SRC_DIR}
+    )
+    message(STATUS "Fetching Boost from official release.")
+    FetchContent_MakeAvailable(Boost)
+endif ()
+
+set(BIN_NAME "ClpIrV1Decoder" CACHE STRING "Binary name for the generated .js file and .wasm file of CLP IR V1 Decoder.")
+add_executable(${BIN_NAME})
+
+target_compile_features(${BIN_NAME}
+    PRIVATE
+    cxx_std_20
+)
+
+set(CMAKE_EXECUTABLE_SUFFIX ".js" CACHE STRING "Binary type to be generated by Emscripten")
+
+if (CMAKE_BUILD_TYPE MATCHES "Release")
+    set(EXTRA_LINKER_FLAGS
+        -flto
+        --closure=1
+        -sENVIRONMENT=worker
+        CACHE STRING "Extra liner flags for CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}" FORCE
+    )
+    message("CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}: Enabling optimization linker flags ${EXTRA_LINKER_FLAGS}")
+else ()
+    set(EXTRA_LINKER_FLAGS
+        -sENVIRONMENT=node
+        CACHE STRING "Extra liner flags for CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}" FORCE
+    )
+    message("CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}: Enabling debug flags linker for ${EXTRA_LINKER_FLAGS}")
+endif ()
+target_link_options(${BIN_NAME} PRIVATE
+    ${EXTRA_LINKER_FLAGS}
+    -sALLOW_MEMORY_GROWTH
+    -sEXPORT_ES6
+    -sMODULARIZE
+    -sWASM_BIGINT
+)
+target_link_libraries(${BIN_NAME}
+    PRIVATE
+    embind
+)
+
+target_compile_definitions(${BIN_NAME}
+    PUBLIC
+    SPDLOG_FMT_EXTERNAL=1
+)
+
+target_include_directories(${BIN_NAME}
+    PRIVATE
+    ${BOOST_SRC_DIR}
+    src/
+    src/submodules/clp/components/core/src
+    src/submodules/clp/components/core/src/clp
+    src/submodules/clp/components/core/submodules
+    src/submodules/fmt/include
+    src/submodules/spdlog/include
+    src/submodules/zstd/lib
+)
+
+target_sources(${BIN_NAME}
+    PRIVATE
+    src/ClpIrV1Decoder.cpp
+
+    src/submodules/clp/components/core/src/clp/ffi/ir_stream/decoding_methods.cpp
+    src/submodules/clp/components/core/src/clp/ir/LogEventDeserializer.cpp
+    src/submodules/clp/components/core/src/clp/ReadOnlyMemoryMappedFile.cpp
+    src/submodules/clp/components/core/src/clp/ReaderInterface.cpp
+    src/submodules/clp/components/core/src/clp/streaming_compression/zstd/Decompressor.cpp
+    src/submodules/clp/components/core/src/clp/TimestampPattern.cpp
```

Comment:
As a convention we used in core, we should also include all the header files we used